File: fixtures/kotlin/type_inference/visibility/Visibility.kt
package app.models

open class Visibility {
    val pubField: String = "public"
    internal val internalField: String = "internal"
    protected val protectedField: String = "protected"
    private val privateField: String = "private"

    fun getPublic(): String = pubField

    internal fun getInternal(): String = internalField

    protected fun getProtected(): String = protectedField

    private fun getPrivate(): String = privateField

    fun describe(): String = getPublic() + getInternal() + getProtected() + getPrivate()
}

File: fixtures/scala/type_inference/visibility/Visibility.scala
package app.models

class Visibility {
  val pubField: String = "public"
  private val privateField: String = "private"
  protected val protectedField: String = "protected"

  def getPublic(): String = pubField

  private def getPrivate(): String = privateField

  protected def getProtected(): String = protectedField

  def describe(): String = getPublic() + getPrivate() + getProtected()
}
